Userland/Utilities/pkg/InstalledPort.cpp

```diff
@@ -38,29 +38,22 @@ ErrorOr<HashMap<String, InstalledPort>> InstalledPort::read_ports_database()
             // FIXME: Skip over invalid entries instead?
             return Error::from_string_view("Database entry too short"sv);
         }
-        auto string_type = parts[0];
-        auto name = TRY(String::from_utf8(parts[1]));
+        auto port_name = TRY(String::from_utf8(parts[1]));
 
-        if (auto maybe_type = type_from_string(string_type); maybe_type.has_value()) {
+        if (auto maybe_type = type_from_string(parts[0]); maybe_type.has_value()) {
             auto const type = maybe_type.release_value();
             if (parts.size() < 3)
                 return Error::from_string_view("Port is missing a version specification"sv);
-            auto version = TRY(String::from_utf8(parts[2]));
 
-            auto& port = ports.ensure(name, [&] { return InstalledPort { name, {}, {} }; });
-            port.m_type = type;
-            port.m_version = move(version);
-        } else if (string_type == "dependency"sv) {
-            // Accept an empty dependency list.
-            auto dependency_views = parts.span().slice(2);
+            ports.ensure(port_name, [=] { return InstalledPort { port_name, MUST(String::from_utf8(parts[2])), type }; });
+        } else if (parts[0] == "dependency"sv) {
             Vector<String> dependencies;
-            TRY(dependencies.try_ensure_capacity(dependency_views.size()));
-            for (auto const& view : dependency_views)
-                dependencies.unchecked_append(TRY(String::from_utf8(view)));
-
+            for (auto const& dependency : parts.span().slice(2)) {
+                TRY(dependencies.try_append(TRY(String::from_utf8(dependency))));
+            }
```

Comment:
why did you remove the ensure_capacity + unchecked_append? It's strictly better for memory usage.

Reply:
I didnt know that. Thanks for this. (`Optional<But can you provide some reference and citations to know more about it?>`)